packages/core/core/src/public/Symbols.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+import type {
+  Symbol,
+  MutableSymbols as IMutableSymbols,
+  Symbols as ISymbols,
+  SourceLocation,
+} from '@parcel/types';
+
+let parcelSymbolsToSymbols: WeakMap<
+  Map<Symbol, {|local: Symbol, loc: ?SourceLocation|}>,
+  Symbols,
+> = new WeakMap();
+
+export class Symbols implements ISymbols {
+  #symbols; // Map<Symbol, {|local: Symbol, loc: ?SourceLocation|}>
+
+  constructor(symbols: Map<Symbol, {|local: Symbol, loc: ?SourceLocation|}>) {
+    let existing = parcelSymbolsToSymbols.get(symbols);
+    if (existing != null) {
+      return existing;
+    }
+
+    this.#symbols = symbols;
+    parcelSymbolsToSymbols.set(symbols, this);
+  }
+
+  get(exportSymbol: Symbol): ?{|local: Symbol, loc: ?SourceLocation|} {
+    return this.#symbols.get(exportSymbol);
+  }
+
+  hasExportSymbol(exportSymbol: Symbol): boolean {
+    return this.#symbols.has(exportSymbol);
+  }
+
+  hasLocalSymbol(local: Symbol): boolean {
+    for (let s of this.#symbols.values()) {
+      if (local === s.local) return true;
+    }
+    return false;
+  }
+
+  getAll(): $ReadOnlyMap<Symbol, {|local: Symbol, loc: ?SourceLocation|}> {
```

Comment:
I know the flow type is read only, but it's technically writable because JS. Looks like this is only used to iterate over the symbols. So maybe we implement the iterable interface for Symbols instead? e.g. `[Symbol.iterator]`

Reply:
Done (and it looks cleaner anyway)